lenskit-eval/src/main/java/org/lenskit/eval/traintest/WebAppListener.java

```diff
@@ -0,0 +1,206 @@
+            con.setRequestProperty("User-Agent", USER_AGENT);
+            con.setRequestProperty("Accept-Language", "en-US,en;q=0.5");
+            con.setRequestProperty("Content-Type", "application/json");
+            con.setDoOutput(true);
+            String description = jf.getJob().getDescription();
+            TrackedJob job = jf.getJob();
+            TrackedJob parent = job.getParent();
+            Throwable exception;
+            exception = jf.getException();
+            String jobString = CreateJSON(job, parent, 3);
+            sendPostRequest(jobString, con);
+        }catch(java.net.ConnectException e){
+
+        }
+    }
+
+    /**
+     * EventProgressUpdate is subscribed to the event bus and waits for it to send out a JobEvent.ProgressUpdate object.
+     * The JobEvent information is then parsed into a JSON object and sent to the WebApp server
+     * @param jp : Job Event that gives a progress update of the current job
+     * @throws Exception
+     */
+    @Subscribe
+    public static void EventProgressUpdate(JobEvent.ProgressUpdate jp) throws Exception{
+        try {
+            String url = "http://localhost:3000";
+            URL obj = new URL(url);
+            HttpURLConnection con = (HttpURLConnection) obj.openConnection();
+            con.setRequestMethod("POST");
+            con.setRequestProperty("User-Agent", USER_AGENT);
+            con.setRequestProperty("Accept-Language", "en-US,en;q=0.5");
+            con.setRequestProperty("Content-Type", "application/json");
+            con.setDoOutput(true);
+            TrackedJob job = jp.getJob();
+            TrackedJob parent = job.getParent();
+            String jobString = CreateJSON(job, parent, 1);
+            sendPostRequest(jobString, con);
+        }catch(java.net.ConnectException e){
+
+        }
+    }
+
+    /**
+     * The JSON String is sent to the WebApp server
+     * @param jobInfoJson : The JSON object that will be sent to the server
+     * @param con : The HttpURLConnection to the server.
+     * @throws Exception
+     */
+
+    private static void sendPostRequest(String jobInfoJson, HttpURLConnection con) throws Exception {
+
+        DataOutputStream wr = new DataOutputStream(con.getOutputStream());
+        wr.writeBytes(jobInfoJson);
+        wr.flush();
+        wr.close();
+        BufferedReader in = new BufferedReader( new InputStreamReader(con.getInputStream()));
+        String output;
+        StringBuffer response = new StringBuffer();
+
+        while ((output = in.readLine()) != null) {
+            response.append(output);
+        }
+        in.close();
+
+    }
+
+    /**
+     *
+     * @param job : The job that we are creating the json string for.
+     * @param parent : The parent of the current job
+     * @param typeOfUpdate : 0=Start 1=Progress 2=Finish 3=Failed
+     * @return : Return the json string
+     */
+    private static String CreateJSON(TrackedJob job,TrackedJob parent, int typeOfUpdate){
```

Comment:
This can probably be simplified with Jackson's JSON building capabilities - see some fo our `toJSON` methods for examples.